src/lib/transition_handler/catchup_scheduler.ml

```diff
@@ -30,11 +30,18 @@ module Make (Inputs : Inputs.S) = struct
         , synchronous
         , unit Deferred.t )
         Writer.t
-    ; timeouts:
-        ( (External_transition.Verified.t, State_hash.t) With_hash.t
-        , unit Time.Timeout.t )
-        List.Assoc.t
+    ; collected_transitions:
+        (External_transition.Verified.t, State_hash.t) With_hash.t list
         State_hash.Table.t
+        (* The new implementation has this new hash table storing all seen
```

Comment:
For comments, do mention what is a `new` implementation vs an `old` implementation. Code changes all the time. So, your meaning of `new` might be `old` in the future.